### ISSUE TITLE
MediaInfo displays "Progressive" for  MXF/JPEG2000 with SMPTE 422 I2 wrapping mode

### ISSUE THREAD

**Issue by plondino** [MediaArea/MediaInfo#641]
Mediainfo displays "Progressive" for **Scan type** for an JPEG 2000 in an MXF wrapped with the I2 wrapping mode.  The 
default is "Frame Undefined" (06 0E 2B 34 04 01 01 07 0D 01 03 01 02 0C 01 00) where the frame structure was not specified in the standard. ST 422:2013 defined new wrapping modes, including explicit interlaced modes.  

I2 wrapping mode (06 0E 2B 34 04 01 01 0D 0D 01 03 01 02 0C 04 00) has the below frame structure:

<img width="999" alt="Screen Shot 2022-09-13 at 10 12 45 AM" src="https://user-images.githubusercontent.com/40502255/190052586-7dac00a8-3d42-40a5-a4ea-af930ce9455a.png">

<img width="999" alt="Screen Shot 2022-09-13 at 10 12 45 AM" src="https://user-images.githubusercontent.com/40502255/190052551-2aaf674a-4c54-4d64-8ad9-2acf2d65fd05.png">

MediaInfo should read this as interlaced.  I can provide a sample file.

**Comment by JeromeMartinez** [comment]
> Mediainfo displays "Progressive" for Scan type for an JPEG 2000 in an MXF wrapped with the I2 wrapping mode. 

As far as we understand the spec, the 2 stored fields are expected to be displayed at the same time ("frame" sample rate), so scan type is "progressive" on purpose. Does HW display a different output at 1.5 frame than 1 frame or 2 frames? i.e. at 1.5 frame (0.050 second), 1 field of first frame and 1 field of second frame are displayed and not the 2 fields of first frame like 17 ms sooner?

IMO what is missing in the MediaInfo report is how the frame is stored. We already show a specific fields "Store method" and "Fields per block" for MOV, doing the difference between the different possibilities of "fiel" atom, we could do same for MXF.

**Comment by plondino** [comment]
I reached out to some SMPTE folks about this.  Here is the response I got:

_It looks like mediainfo might be using the FrameLayout property
instead of using the Essence Container UL to determine whether the
video is interlaced: in the case of I2, and as specified in ST 422,
Table A.1, FrameLayout is set to FullFrame even though the video is
interlaced._

This is my understanding as well - the old "Frame Undefined" wrapping was ambiguous and most implementations keyed off the FRAME_LAYOUT property.  The new document defines three interlace wrapping modes (I1, I2, and F1) which are all interlaced content with different frame storage options.  Progressive content should be wrapped with the P1 mode.

I do think having the store method is useful as well, but it doesn't supersede the fact that I2 wrapping is in fact interlaced.

**Comment by plondino** [comment]
You're correct, the file I sent should have had the StoredHeight of FULL_FRAME.  I will send you an update file which is correct*.

* I am still a little unclear on what the VideoLineMap property is supposed to be for I2 - ST 422:2019 says:

_SMPTE ST 377-1 Annex F.4 and Annex G give normative provisions for specifying the values of other properties
of the Picture Essence Descriptor, including Video Line Map, Stored Height and Width, Sampled Height and
Width, and Display Height and Width. The correspondence between the values of these and frame or field is
implied by the value of the FrameLayout property, as tabulated in SMPTE ST 377-1 Annex G.2.7, G.2.9 and
G.2.14._

I read the table from SMPTE ST 377-1:2019 as implying that the value for full frame should be 42,0 - but I also feel like the VideoLineMap indicates where the video was sample _from_ which would be lines 21,582.  I may have to reach out to someone at SMPTE for clarification.
 
<img width="833" alt="videolinemap" src="https://github.com/MediaArea/MediaInfo/assets/40502255/37cd3a3e-32ce-4263-a7ff-2689e377eb29">
